metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -619,6 +633,22 @@ def construct_dataset_from_dataset_data(
     def emit_dataset_mces(self) -> Iterable[MetadataWorkUnit]:
         for dataset_data in self.paginate_entity_api_results("dataset", PAGE_SIZE):
             try:
+                dataset_id = dataset_data.get("id")
+                dataset_response = self.get_dataset_info(dataset_id)
+
+                database_name = (
+                    dataset_response.get("result", {})
+                    .get("database", {})
+                    .get("database_name", "")
+                )
```

Comment:
if this logic is happening in multiple places, can we extract it out into a helper method?